crates/btc-relay/src/lib.rs

```diff
@@ -1094,21 +1094,11 @@ impl<T: Config> Pallet<T> {
 
                 // update the storage
                 Self::mutate_block_chain_from_id(new_fork.chain_id, new_fork.clone());
-
-                // if we reached main chain, we need to move all blocks that are no longer to be
-                // in the mainchain to a normal fork
-                if parent.chain_ref == MAIN_CHAIN_ID {
-                    for height in new_fork.start_height..=new_fork.max_height {
-                        let block_hash = Self::get_block_hash(MAIN_CHAIN_ID, height)?;
-                        let block = Self::get_block_header_from_hash(block_hash)?;
-
-                        Self::transfer_block_to_chain(block, MAIN_CHAIN_ID, fork.chain_id)?;
-                    }
-                }
             }
 
-            // transfer the child block to the main chain
-            Self::transfer_block_to_chain(child, child.chain_ref, MAIN_CHAIN_ID)?;
+            // transfer the child block to the main chain, and if main already has a block at this
+            // height, transfer it to `fork`
+            Self::swap_block_to_mainchain(child, child.chain_ref, fork.chain_id)?;
```

Comment:
```suggestion
            Self::swap_block_to_mainchain(child, fork.chain_id)?;
```

Reply:
Actually it's not clear to me why this is even called `chain_ref`, can this be renamed to `chain_id`?

Reply:
Good point, renamed.